src/responder/secrets/local.c

```diff
@@ -373,13 +373,41 @@ int local_db_delete(TALLOC_CTX *mem_ctx,
                     const char *req_path)
 {
     struct ldb_dn *dn;
+    static const char *attrs[] = { NULL };
+    struct ldb_result *res = NULL;
     int ret;
 
     ret = local_db_dn(mem_ctx, lctx->ldb, req_path, &dn);
     if (ret != EOK) return ret;
 
-    ret = ldb_delete(lctx->ldb, dn);
-    return sysdb_error_to_errno(ret);
+    ret = ldb_search(lctx->ldb, mem_ctx, &res, dn, LDB_SCOPE_BASE,
+                    attrs, LOCAL_CONTAINER_FILTER);
+    if (ret != EOK) return ret;
+
+    if (res->count == 1) {
+        talloc_free(res);
```

Comment:
You should use a tmp_ctx in this function, and allocate the dn above, and all searches on it, then just talloc_free() the tmp_ctx at the end.